File: 0101-0200/0167.Two_Sum_II-Input_Array_Is_Sorted.java
//2ms




class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int left = 0;
        int right = numbers.length - 1;
        while (left < right) {
            int sum = numbers[left] + numbers[right];
            if (sum == target) {
                return new int[] { left + 1, right + 1 };
            } else if (sum < target) {
                left++;
            } else {
                right--;
            }
        }
        return new int[] { -1, -1 };
    }
}






//0ms






class Solution {
    static {
        int[] numbers = {0,1};
        for (int i=0; i<500; i++) {
            twoSum(numbers, 1);
        }
    }
    public static int[] twoSum(int[] numbers, int target) {
        int start = 0;
        int end = numbers.length-1;

        while (start<end) {
            int sum = numbers[start] + numbers[end];
            if (sum==target) return new int[]{start+1,end+1};
            if (sum<target) start++;
            if (sum>target) end--;
        }

        return new int[]{0,0};
    }
}

File: 0101-0200/0167.Two_Sum_II-Input_Array_Is_Sorted.js
//2m s





/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (numbers, target) {
    let i = 0;
    let j = numbers.length - 1;
    while (j > i) {
        if (numbers[i] + numbers[j] === target) {
            return [i+1,j+1]
        } else if (numbers[i] + numbers[j] > target){
            j--;
        } else {
            i++
        }
    }
};





//1ms





/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (numbers, target) {
    let left = 0;
    let right = numbers.length - 1;

    while (left < right) {
        const sum = numbers[left] + numbers[right];
        if (sum === target) {
            return [left + 1, right + 1];
        }
        else if (sum < target) {
            left++;
        }
        else {
            right--;
        }
    }
};
